tests/std/tests/GH_001858_iostream_exception/test.cpp

```diff
@@ -148,6 +149,254 @@ void test_istream_exceptions() {
+        is.exceptions(ios_base::failbit);
+        assert(!is.bad());
+
+        // tests null termination on eof and
+        // sets eofbit, preparing sentry failure
+        auto buf1 = meow_array<CharT>;
+        is.get(buf1.data(), static_cast<streamsize>(buf1.size()));
+        assert(is.eof());
+        assert(!is.fail());
+        assert(buf1[0] == meow_array<CharT>[2]);
+        assert(buf1[1] == CharT());
+
+        // actually tests sentry construction failure
+        auto buf2 = meow_array<CharT>;
+        try {
+            is.get(buf2.data(), static_cast<streamsize>(buf2.size()));
+            assert(false);
+        } catch (const ios_base::failure&) {
+            // Expected case
+        }
+        assert(is.fail());
+        assert(buf2[0] == CharT());
+    }
+
+    { // get, exception raised on eofbit
+        basic_stringbuf<CharT> strbuf{stream_content};
+        basic_istream<CharT> is(&strbuf);
+        is.exceptions(ios_base::eofbit);
+        assert(!is.bad());
+
+        auto buf = meow_array<CharT>;
+        try {
+            is.get(buf.data(), static_cast<streamsize>(buf.size()));
+            assert(false);
+        } catch (const ios_base::failure&) {
+            // Expected case
+        }
+        assert(is.eof());
+        assert(!is.fail());
+        assert(buf[0] == meow_array<CharT>[2]);
+        assert(buf[1] == CharT());
+    }
+
+    { // getline, exception during input extraction, no exception rethrow
+        basic_istream<CharT> is(buffer.to_buf());
+        auto buf = meow_array<CharT>;
+        assert(!is.bad());
+        is.getline(buf.data(), static_cast<streamsize>(buf.size()));
+        assert(is.bad());
+        assert(buf[0] == CharT());
+    }
+
+    { // getline, exception during input extraction, exception rethrow enabled
+        basic_istream<CharT> is(buffer.to_buf());
+        auto buf = meow_array<CharT>;
+        is.exceptions(ios_base::badbit);
+        assert(!is.bad());
+        try {
+            is.getline(buf.data(), static_cast<streamsize>(buf.size()));
+            assert(false);
+        } catch (const ios_base::failure&) {
+            assert(false);
+        } catch (const test_exception&) {
+            // Expected case
+        }
+        assert(is.bad());
+        assert(buf[0] == CharT());
+    }
+
+    { // getline, empty output buffer, no exception raised
+        basic_istream<CharT> is(buffer.to_buf());
+        auto buf = meow_array<CharT>;
+        assert(!is.bad());
+        is.getline(buf.data(), 0);
+        assert(is.fail());
+        assert(buf[0] == meow_array<CharT>[0]);
+    }
+
+    { // getline, empty output buffer, exception raised on failbit
+        basic_istream<CharT> is(buffer.to_buf());
+        auto buf = meow_array<CharT>;
+        is.exceptions(ios_base::failbit);
+        assert(!is.bad());
+        try {
+            is.getline(buf.data(), 0);
+        } catch (const ios_base::failure&) {
+            // Expected case
+        }

```

Comment:
Diffing `get` versus `getline` reveals that there's an `assert(false)` missing from the no-exception-thrown codepath here.